src/pages/links/index.tsx

```diff
@@ -1,6 +1,488 @@
+      ...updatedQuery,
+    }));
+  }, []);
+  return { query, update };
+};
+
+const useLinksQueryHook = (width: number, folderId: number = 0) => {
+  const [query, setQuery] = useState(initialLinksQueryState(width));
+  const update = useCallback(
+    (updatedQuery: Partial<LinksQuery>) => {
+      setQuery((prevQuery) => ({
+        ...prevQuery,
+        ...updatedQuery,
+        id: folderId,
+      }));
+    },
+    [folderId],
+  );
+  useEffect(() => {
+    update({ pageSize: getPageSize(width) });
+  }, [update, width]);
+  return { query, update };
+};
+
+type LinkListPagenationComponentProps = {
+  query: LinksQuery;
+  width: number;
+  update: (updatedQuery: Partial<LinksQuery>) => void;
+};
+
+const getPagination = (page: number, totalPages: number, width: number) => {
+  if (totalPages <= 7) {
+    return Array.from({ length: totalPages }, (_, i) => i + 1);
+  }
+
+  const isNarrowScreen = width < 1024;
+  const isVeryNarrowScreen = width < 425;
+  const nearStart = page <= 3;
+  const nearEnd = page >= totalPages - 2;
+
+  if (nearStart) {
+    return isNarrowScreen
+      ? [1, 2, 3, '...', totalPages]
+      : [1, 2, 3, '...', totalPages - 1, totalPages];
+  }
+
+  if (nearEnd) {
+    return isNarrowScreen
+      ? [1, '...', totalPages - 2, totalPages - 1, totalPages]
+      : [1, 2, '...', totalPages - 2, totalPages - 1, totalPages];
+  }
+
+  return isVeryNarrowScreen
+    ? [page - 2, page - 1, page, page + 1, page + 2]
+    : [1, '...', page - 1, page, page + 1, '...', totalPages];
+};
+
+type PagenationNumberComponentProps = {
+  page: number | string;
+  position: 'first' | 'last' | 'middle' | 'single' | undefined;
+  isActive: boolean;
+  query: Partial<LinksQuery> | undefined;
+  update: (updatedQuery: Partial<LinksQuery>) => void;
+};
+const pageStyle =
+  'box-content w-7 h-7 p-2 text-center rounded-lg bg-gray-200 text-lg leading-relaxed';
+const PaginationNumberComponent = ({
+  page,
+  isActive,
+  position,
+  query,
+  update,
+}: PagenationNumberComponentProps) => {
+  return isActive || position === 'middle' ? (
+    <li className={`${pageStyle} font-semibold`}>{page}</li>
+  ) : (
+    <li className={pageStyle}>
+      <button
+        className="w-full h-full"
+        onClick={() => update(query as LinksQuery)}
+      >
+        {page}
+      </button>
+    </li>
+  );
+};
+
+type PagenationArrowComponentProps = {
+  query: Partial<LinksQuery>;
+  direction: 'left' | 'right';
+  isDisabled: boolean;
+  update: (updatedQuery: Partial<LinksQuery>) => void;
+};
+const PagenationArrowComponent = ({
+  query,
+  direction,
+  isDisabled,
+  update,
+}: PagenationArrowComponentProps) => {
+  const arrow = direction === 'left' ? '<' : '>';
+  return isDisabled ? (
+    <li className={pageStyle}>
+      <button className="w-full h-full">{arrow}</button>
+    </li>
+  ) : (
+    <li className={`${pageStyle} bg-green-600 text-white`}>
+      <button onClick={() => update(query)}>{arrow}</button>
+    </li>
+  );
```

Comment:
`ts-pattern`을 적용해보시는건 어떠신가요? 조금 더 가독성 있는 코드가 될 수 있을 것 같습니다! 😄

Reply:
혹은 `button`의 attribute인 `disabled`를 활용하는 것도 좋아보입니다!
```suggestion
  return (
    <li className={pageStyle}>
      <button className="w-full h-full" onClick={() => update(query)} disabled={isDisabled}>{arrow}</button>
    </li>
  );
```